### PR TITLE
Fix potential overflow in sys_vm

### PR DESCRIPTION
**Pull request by elad335**

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_vm.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ error_code sys_vm_return_memory(u32 addr, u32 size)
 
 	std::lock_guard lock(block->mutex);
 
-	if (block->psize < 0x100000 + size)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_vm.cpp]: Using functional cast isn't good, and there is also no point in replacing < with >=.
u64{x} can be used instead, and also check the rules for integer promotion. It's only necessary to promote only one part of the expression to u64, the other part will promote implicitly.

File: rpcs3/Emu/Cell/lv2/sys_vm.cpp [hunk 2]
```diff
@@ -156,7 +156,7 @@ error_code sys_vm_lock(u32 addr, u32 size)
 
 	const auto block = idm::get<sys_vm_t>(sys_vm_t::find_id(addr));
 
-	if (!block || addr + size > block->addr + block->size)
+	if (!block || (u64)addr + (u64)size > (u64)block->addr + (u64)block->size)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_vm.cpp]: `u64{addr} + size > u64{block->addr} + block->size` should be enough.

**Comment by stride21** [comment]
Beyond Two Souls is getting an error on start up.

E {PPU[0x1000000] Thread (main_thread) [0x00c15cb0]} sys_vm: sys_vm_memory_map(vsize=0x10000000, psize=0xbc00000, cid=0x1, flags=0x200, policy=0x1, addr=*0x11f778c)

F {PPU[0x1000000] Thread (main_thread) [0x00c15cb0]} class std::runtime_error thrown: Verification failed: 
(in file c:\projects\rpcs3\rpcs3\emu\cell\lv2\sys_vm.cpp:53)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/3316396/RPCS3.log.gz)

**Comment by RainbowCookie32** [comment]
Someone else on Discord also reports the same error, apparently it started happening after this PR was merged

[RPCS3 (2).log.gz](https://github.com/RPCS3/rpcs3/files/3316423/RPCS3.2.log.gz)

**Comment by elad335** [comment]
Retest with #6120

**Pull request merged**